src/linear_algebra/nonsingular_inverse.lean

```diff
@@ -354,33 +354,76 @@ Defines the matrix `nonsing_inv A` and proves it is the inverse matrix
 of a square matrix `A` as long as `det A` has a multiplicative inverse.
 -/
 
-variables [comm_ring α] [has_inv α]
+variables [comm_ring α]
+variables (A : matrix n n α)
 
-/-- The inverse of a nonsingular matrix.
+open_locale classical
 
-  This is not the most general possible definition, so we don't instantiate `has_inv` (yet).
```

Comment:
I don't know what happened to this comment. Do you want to declare a `has_inv` instance, or not yet?

Reply:
Personally I'm in favour of adding the instance. I refrained from doing so only because I had not fulfilled the intentions of the original author. (I guess it's a bit inconsistent of me to remove the comment you highlight, given this.)

I think it would only take the slightest encouragement to get me to add the instance. What do you reckon?

Reply:
I'd say add the instance. If we want to change it for a more general definition later, we can do it when needed.

Reply:
Done!